lib/winappdriver.js

```diff
@@ -130,8 +132,37 @@ class WinAppDriver extends events.EventEmitter {
   }
 
   async startSession (caps) {
+
+    let createSessionTimeout = caps.createSessionTimeout || this.createSessionTimeout;
+    log.info(`Start WinAppDriver session during createSessionTimeout = '${createSessionTimeout}' ms.`);
     this.proxyReqRes = this.jwproxy.proxyReqRes.bind(this.jwproxy);
-    await this.jwproxy.command('/session', 'POST', {desiredCapabilities: caps});
+    let retryIteration = 0;
+    let lastError = null;
+
+    const condFn = async () => {
+      try {
+        retryIteration++;
+        await this.jwproxy.command('/session', 'POST', {desiredCapabilities: caps});
+        return true;
+      } catch (error) {
+        lastError = error;
+        log.warn(`Could not start WinAppDriver session error = '${error}', attempt = '${retryIteration}' from '${this.createSessionRetry}'`);
+        return false;
+      }
+    };
+
+    try {
+      await waitForCondition(condFn, {
+        waitMs: createSessionTimeout,
+        intervalMs: 500
+      });
+    } catch (timeoutError) {
+      log.debug(timeoutError);
+      if (lastError) {
+        throw (lastError);
+      }
+      throw (timeoutError);
```

Comment:
the native error thrown by `waitForCondition` is not very informative. We could provide a better error message here